crates/ns_lookup/src/lib.rs

```diff
@@ -26,10 +27,47 @@ use trust_dns_resolver::{
 pub enum Error {
     #[snafu(display("Failed to connect to {host}:{port}, are the host and port correct?"))]
     UnableToConnect { host: String, port: u16 },
+
+    #[snafu(display("Failed to parse endpoint {endpoint}: {source}"))]
+    UnableToParseUrl {
+        endpoint: String,
+        source: url::ParseError,
+    },
+
```

Comment:
it is unlikely, but still possible due to types, for example there is invalid character in the url specified(typo): `grpc://20 .163.171.81`

>Failed to parse endpoint grpc://20 .163.171.81: invalid domain character

Reply:
For ports - there is no default port for grpc:// in comparison to http:// and https://.  For gRPC, the use of port 50051 is more of a convention rather than a formal standard.

We use [URL](https://docs.rs/url/latest/url/) create to parse and the see if there is default port available for schema.

My reasoning here is that we can fallback on some defaults (80 for example) but this it not how connection will be established after that (if this is not standardized) and it will be harder to detect issue later.